system/lib/wasmfs/file_table.h

```diff
@@ -21,15 +21,16 @@ std::shared_ptr<Directory> getRootDirectory();
 
 class OpenFileState : public std::enable_shared_from_this<OpenFileState> {
   std::shared_ptr<File> file;
-  __wasi_filedelta_t offset;
+  size_t position;
+  uint32_t flags; // RD_ONLY, WR_ONLY, RDWR
```

Comment:
Should we add a typedef for the flags? That seems better than `uint32_t` around the codebase, which later might be annoying to refactor.